### PR TITLE
Nikolai/feature/serial lookup

### PR DESCRIPTION
**Pull request by lizard-52**
You can get a description from a serial number now. Solution is a bit messy (calling fragments context gets passed around) but I couldn't find a better library to do this with.
Using this api:
https://devs.upcitemdb.com/

### CODE REVIEW

**Review by waridh — COMMENTED** [review]
I am alright with the code, but I don't think I am in a position to approve since I don't have an android phone to test this with.

**Review by echeng-git — APPROVED** [review]
Looks good to merge

File: code/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
         android:name="android.hardware.camera"
         android:required="false" />
     <uses-permission android:name="android.permission.CAMERA"/>
-    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
+    <uses-permission android:name="android.permission.INTERNET"/>
```
Comment (Yunyang-Lu) [inline code/app/src/main/AndroidManifest.xml:9]: external storage is needed then taking gallery photo, please keep it

Comment (lizard-52) [reply code/app/src/main/AndroidManifest.xml:9]: Fixed now

**Comment by lizard-52** [comment]
Also added a fix to gradle scripts. App should build more reliably now.

**Review by JoshuaShaw-13 — APPROVED** [review]
I see no problems, it works for me on my phone

**Pull request merged**